### PR TITLE
i18n: Adds German translation for vocabularies

### PR DESCRIPTION
**Pull request by schranzhofer**
:heart: Thank you for your contribution!

### Description

Translation to German in the following files:
community_types.yaml
date_types.yaml
description_types.yaml
relation_types.yaml
resource_types.yaml
title_types.yaml

### Checklist

I only did some translations. Nothing else.

### CODE REVIEW

**Comment by chriz-uniba** [comment]
- these files do not have any copyright note, would it make sense to add information about translators, etc.?

- I heard elsewhere that yaml-files usually start with `---` should we do that as well? https://yaml.org/refcard.html
  ```
  Document indicators:
    '%'  : Directive indicator.
    '---': Document header.
    '...': Document terminator.
  ```

**Comment by lnielsen** [comment]
Just a quick comment:

- Copyright notice makes sense and yaml should be able to have comments.
- Regarding adding translators, then it depends on the purpose. For credit I would just list them in AUTHORS file (https://inveniordm.docs.cern.ch/contribute/copyright-policy/#tracking-copyright). If it's meaningful for some other purpose (e.g. who is maintaining the file or something similar, then it can go directly in the file).

File: invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml [hunk 1]
```diff
@@ -3,153 +3,184 @@
...
      datacite: Cites
   title:
      en: Cites
+     de: Zitiert
 - id: issupplementto
   props:
      datacite: IsSupplementTo
   title:
      en: Is supplement to
+     de: Ist eine Ergänzung zu
```
Comment (Herrner) [inline invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml:18]: ergänzt? (analog zu "beschreibt")

File: invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml [hunk 2]
```diff
@@ -3,153 +3,184 @@
...
      datacite: IsVariantFormOf
   title:
      en: Is variant form of
+     de: Ist eine Variante von
 - id: isoriginalformof
   props:
      datacite: IsOriginalFormOf
   title:
      en: Is original form of
+     de: Ist die Orginale von
```
Comment (Herrner) [inline invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml:132]: `Ist das Original von` oder `Ist die Originalform von`

File: invenio_rdm_records/fixtures/data/vocabularies/resource_types.yaml [hunk 3]
```diff
@@ -250,6 +264,7 @@
     type: publication
   title:
     en: Taxonomic treatment
+    de: Steuerliche Abhandlung
```
Comment (Herrner) [inline invenio_rdm_records/fixtures/data/vocabularies/resource_types.yaml:267]: `Taxonomische Behandlung` (auf keinen Fall "steuerlich")

**Review by Herrner — CHANGES_REQUESTED** [review]

File: invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml (hunk 2 above)
Comment (utnapischtim) [reply invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml:132]: I think `Ist das Original von` is better

File: invenio_rdm_records/fixtures/data/vocabularies/resource_types.yaml (hunk 3 above)
Comment (chriz-uniba) [reply invenio_rdm_records/fixtures/data/vocabularies/resource_types.yaml]: Thanks for checking @Herrner

File: invenio_rdm_records/fixtures/data/vocabularies/resource_types.yaml [hunk 4]
```diff
@@ -165,6 +174,7 @@
     type: publication
   title:
     en: Project deliverable
+    de: Projektergebnis
```
[the remark this answers is not in the document]
Comment (chriz-uniba) [reply invenio_rdm_records/fixtures/data/vocabularies/resource_types.yaml:177]: Decided: okay

File: invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml [hunk 5]
```diff
@@ -3,153 +3,184 @@
...
      datacite: Requires
   title:
      en: Requires
+     de: Voraussetzungen
 - id: isobsoletedby
   props:
      datacite: IsObsoletedBy
   title:
      en: Is obsoleted by
+     de: Wird ersetzt durch
```
[the remark this answers is not in the document]
Comment (chriz-uniba) [reply invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml:180]: Decided: okay

File: invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml [hunk 6]
```diff
@@ -3,153 +3,184 @@
...
      datacite: IsDerivedFrom
   title:
      en: Is derived from
+     de: Wird entlehnt aus
 - id: issourceof
   props:
      datacite: IsSourceOf
   title:
      en: Is source of
+     de: Ist Quelle von
```
[the remark this answers is not in the document]
Comment (chriz-uniba) [reply invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml:162]: Decided: okay

File: invenio_rdm_records/fixtures/data/vocabularies/description_types.yaml [hunk 7]
```diff
@@ -3,28 +3,34 @@
...
     datacite: TechnicalInfo
   title:
     en: Technical info
+    de: Technische Informationen
 - id: other
   props:
     datacite: Other
   title:
     en: Other
+    de: Sonstige
```
[the remark this answers is not in the document]
Comment (chriz-uniba) [reply invenio_rdm_records/fixtures/data/vocabularies/description_types.yaml:36]: Decided: okay

File: invenio_rdm_records/fixtures/data/vocabularies/date_types.yaml [hunk 8]
```diff
@@ -3,53 +3,64 @@
...
     datacite: Created
   title:
     en: Created
+    de: Erzeugt
 - id: issued
   props:
     datacite: Issued
   title:
     en: Issued
+    de: Veröffentlicht
```
[the remark this answers is not in the document]
Comment (chriz-uniba) [reply invenio_rdm_records/fixtures/data/vocabularies/date_types.yaml:36]: Decided: okay

**Review by chriz-uniba — COMMENTED** [review]
I guess the best way to go on is to rebase, clean up commits and merge this. Then we can see it in the instance and work further on that.

File: invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml (hunk 1 above)
Comment (chriz-uniba) [reply invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml]: Fixed

File: invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml (hunk 2 above)
Comment (chriz-uniba) [reply invenio_rdm_records/fixtures/data/vocabularies/relation_types.yaml]: Fixed

**Pull request merged**